fimax/loans/doctype/loan/loan.py

```diff
@@ -66,7 +72,10 @@ def before_cancel(self):
 		self.update_status(new_status="Cancelled")
 
 	def on_cancel(self):
-		pass
+		l_record = frappe.get_doc("Loan Record", self.name)
```

Comment:
This will break. If the Loan Record did not exist the get_doc function will throw an execption. Need be worked!